src/elements/content-sidebar/SidebarPanels.js

```diff
@@ -68,124 +71,167 @@ const LoadableVersionsSidebar = SidebarUtils.getAsyncSidebarContent(
     MARK_NAME_JS_LOADING_VERSIONS,
 );
 
-const SidebarPanels = ({
-    activitySidebarProps,
-    currentUser,
-    detailsSidebarProps,
-    file,
-    fileId,
-    getPreview,
-    getViewer,
-    hasActivity,
-    hasDetails,
-    hasMetadata,
-    hasSkills,
-    hasVersions,
-    isOpen,
-    metadataSidebarProps,
-    onVersionChange,
-    onVersionHistoryClick,
-    refreshIdentity,
-    versionsSidebarProps,
-}: Props) =>
-    isOpen && (
-        <Switch>
-            {hasSkills && (
-                <Route
-                    exact
-                    path={`/${SIDEBAR_VIEW_SKILLS}`}
-                    render={() => (
-                        <LoadableSkillsSidebar
-                            key={file.id}
-                            file={file}
-                            getPreview={getPreview}
-                            getViewer={getViewer}
-                            startMarkName={MARK_NAME_JS_LOADING_SKILLS}
+class SidebarPanels extends React.Component<Props> {
+    activitySidebar: ElementRefType = React.createRef();
+
+    detailsSidebar: ElementRefType = React.createRef();
+
+    metadataSidebar: ElementRefType = React.createRef();
+
+    versionsSidebar: ElementRefType = React.createRef();
+
+    /**
+     * Refreshes the contents of the active sidebar
+     * @returns {void}
+     */
+    refresh(): void {
+        const { current: activitySidebar } = this.activitySidebar;
+        const { current: detailsSidebar } = this.detailsSidebar;
+        const { current: metadataSidebar } = this.metadataSidebar;
+        const { current: versionsSidebar } = this.versionsSidebar;
+
+        if (activitySidebar) {
+            activitySidebar.refresh();
+        }
+
+        if (detailsSidebar) {
+            detailsSidebar.refresh();
+        }
+
+        if (metadataSidebar) {
+            metadataSidebar.refresh();
+        }
+
+        if (versionsSidebar) {
+            versionsSidebar.refresh();
+        }
+    }
+
+    render() {
+        const {
+            activitySidebarProps,
+            currentUser,
+            detailsSidebarProps,
+            file,
+            fileId,
+            getPreview,
+            getViewer,
+            hasActivity,
+            hasDetails,
+            hasMetadata,
+            hasSkills,
+            hasVersions,
+            isOpen,
+            metadataSidebarProps,
+            onVersionChange,
+            onVersionHistoryClick,
+            versionsSidebarProps,
+        }: Props = this.props;
+
+        if (isOpen) {
```

Comment:
Can we make this a guard clause to avoid the level of indentation it introduces?

```
if (!isOpen) {
    return null;
}
```